### PR TITLE
Fix stream method & export as default

### PR DESCRIPTION
**Pull request by balthazar**

### CODE REVIEW

**Comment by hlolli** [comment]
Nice pr. Made me realize how bad the naming is for the functions. To make that error message more relevant, the non boolean returning function "hasEnough" could be renamed to "readBytes" or something, as well. Otherwise lgtm, if @joshbenaron doesn't beat me to a merge, I'll take another look tomorrow!

**Comment by balthazar** [comment]
Yeah `readBytes` makes way more sense 😄

File: stream/index.ts [hunk 1]
```diff
@@ -174,16 +174,18 @@ export async function verifyAndIndexStream(
...
   if (buffer.byteLength > length) return buffer;
-  buffer = Buffer.concat([buffer, (await reader.next()).value]);
-  if (buffer.byteLength > length) return buffer;
 
-  return buffer;
+  const { done, value } = await reader.next();
+
+  if (done && buffer.byteLength < length) throw new Error("Invalid buffer");
+
+  return readBytes(reader, Buffer.concat([buffer, value]), length);
```
Comment (hlolli) [inline stream/index.ts:188]: Correct me if I'm missing out the logic here, but wont this result in recursion until error is thrown? It wasn't recursive before?

The name is good, apologies for the delay, just need to understand :)

**Comment by joshbenaron** [comment]
@JesseTheRobot Can you merge - check all is good with version number, structure etc etc and make new release

**Pull request merged**